// @vonage/cli-plugin-numbers/dist/commands/numbers/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const number_base_1 = tslib_1.__importDefault(require("../../number_base"));
const cli_ux_1 = tslib_1.__importDefault(require("cli-ux"));
class NumberList extends number_base_1.default {
    run() {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const flags = this.parsedFlags;
            let numberData = yield this.getAllNumbers({});
            cli_ux_1.default.table(numberData.numbers, {
                country: {},
                msisdn: {
                    header: "Number"
                },
                type: {},
                features: {
                    get: (row) => row.features.join(',')
                },
                app_id: {
                    header: "Application",
                    get: (row) => row.app_id || ""
                }
            }, Object.assign({}, flags));
        });
    }
    catch(error) {
        const _super = Object.create(null, {
            catch: { get: () => super.catch }
        });
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return _super.catch.call(this, error);
        });
    }
}
exports.default = NumberList;
NumberList.description = 'manage your Vonage numbers';
NumberList.examples = [
    `vonage number`,
];
NumberList.flags = Object.assign(Object.assign({}, number_base_1.default.flags), cli_ux_1.default.table.flags({
    except: ['columns', 'no-truncate', 'csv']
}));
NumberList.args = [];
