// @vonage/cli-plugin-applications/dist/commands/apps/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const app_base_1 = tslib_1.__importDefault(require("../../app_base"));
const cli_ux_1 = tslib_1.__importDefault(require("cli-ux"));
class ApplicationsList extends app_base_1.default {
    run() {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const flags = this.parsedFlags;
            let appData = yield this.allApplications;
            let appList = appData['_embedded'].applications;
            cli_ux_1.default.table(appList, {
                name: {},
                id: {},
                capabilities: {
                    get: (row) => Object.keys(row['capabilities']).toString(),
                }
            }, Object.assign({}, flags));
            this.exit();
        });
    }
    catch(error) {
        const _super = Object.create(null, {
            catch: { get: () => super.catch }
        });
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return _super.catch.call(this, error);
        });
    }
}
exports.default = ApplicationsList;
ApplicationsList.description = 'manage your Vonage applications';
ApplicationsList.examples = ['vonage apps', 'vonage apps --output=json'];
ApplicationsList.flags = Object.assign(Object.assign({}, app_base_1.default.flags), cli_ux_1.default.table.flags({
    except: ['columns', 'no-truncate', 'csv', 'extended', 'no-header']
}));
